django/db/models/fields/json.py

```diff
@@ -373,6 +373,69 @@ class JSONIContains(CaseInsensitiveMixin, lookups.IContains):
     pass
 
 
+class JSONIn(lookups.In):
+    def resolve_expression_parameter(self, compiler, connection, sql, param):
+        sql, params = super().resolve_expression_parameter(
+            compiler,
+            connection,
+            sql,
+            param,
+        )
+        if (
+            not hasattr(param, "as_sql")
+            and not connection.features.has_native_json_field
+        ):
+            if connection.vendor == "oracle":
+                value = json.loads(param)
+                sql = "%s(JSON_OBJECT('value' VALUE %%s FORMAT JSON), '$.value')"
+                if isinstance(value, (list, dict)):
+                    sql %= "JSON_QUERY"
+                else:
+                    sql %= "JSON_VALUE"
+            elif connection.vendor == "mysql" or (
+                connection.vendor == "sqlite"
+                and params[0] not in connection.ops.jsonfield_datatype_values
+            ):
+                sql = "JSON_EXTRACT(%s, '$')"
+        if connection.vendor == "mysql" and connection.mysql_is_mariadb:
+            sql = "JSON_UNQUOTE(%s)" % sql
+        return sql, params
+
+    def process_lhs(self, compiler, connection):
+        sql, params = super().process_lhs(compiler, connection)
+        if isinstance(self.lhs, KeyTransform):
+            return sql, params
+        if connection.vendor in "mysql":
+            sql = "JSON_EXTRACT(%s, '$')" % sql
+        elif connection.vendor == "oracle":
+            if connection.features.supports_primitives_in_json_field:
+                template = (
+                    "COALESCE("
+                    "JSON_VALUE(%s, '$'),"
+                    "JSON_QUERY(%s, '$' DISALLOW SCALARS)"
+                    ")"
```

Comment:
This looks subtly different from the logic in `KeyTransform.as_oracle`. Can you share more about how you derived this code, or why we wouldn't factor out the duplication?